bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitCharacteristicFactory.java

```diff
@@ -614,13 +616,83 @@ private static SaturationCharacteristic createSaturationCharacteristic(HomekitTa
 
     private static ColorTemperatureCharacteristic createColorTemperatureCharacteristic(HomekitTaggedItem taggedItem,
             HomekitAccessoryUpdater updater) {
-        int minValue = taggedItem.getConfigurationAsInt(HomekitTaggedItem.MIN_VALUE,
-                ColorTemperatureCharacteristic.DEFAULT_MIN_VALUE);
-        return new ColorTemperatureCharacteristic(minValue,
-                taggedItem.getConfigurationAsInt(HomekitTaggedItem.MAX_VALUE,
-                        ColorTemperatureCharacteristic.DEFAULT_MAX_VALUE),
-                getIntSupplier(taggedItem, minValue), setIntConsumer(taggedItem),
-                getSubscriber(taggedItem, COLOR_TEMPERATURE, updater),
+        // Check if units are expressed in Kelvin, not mireds, and adjust
+        // the min/max appropriately
+        Unit unit = null;
+        var numberItem = taggedItem.getBaseItem();
+        if (numberItem instanceof NumberItem) {
+            unit = ((NumberItem) numberItem).getUnit();
+        }
+        if (unit == null) {
+            unit = Units.MIRED;
+        }
+        final Unit finalUnit = unit;
+
+        final boolean inverted = taggedItem.isInverted();
+
+        if (!unit.equals(Units.KELVIN) && !unit.equals(Units.MIRED)) {
+            logger.warn("Item {} must be in either K or mired. Given {}.", taggedItem.getName(), unit);
+            return new ColorTemperatureCharacteristic(null, null, null, null);
+        }
+
+        var minValueQt = taggedItem.getConfigurationAsQuantity(HomekitTaggedItem.MIN_VALUE,
+                Objects.requireNonNull(new QuantityType(ColorTemperatureCharacteristic.DEFAULT_MIN_VALUE, Units.MIRED)
+                        .toInvertibleUnit(unit)));
+        var maxValueQt = taggedItem.getConfigurationAsQuantity(HomekitTaggedItem.MAX_VALUE,
+                Objects.requireNonNull(new QuantityType(ColorTemperatureCharacteristic.DEFAULT_MAX_VALUE, Units.MIRED)
+                        .toInvertibleUnit(unit)));
+
+        int minValue = minValueQt.toInvertibleUnit(Units.MIRED).intValue();
```

Comment:
Why conerting first into the item unit and finally into mired? 
Why not directly converting into mired?

Reply:
it's converting the min (in mireds) to the unit of the item, so that the configuration method can know which unit to interpret the value as if no unit is given. then we need to configured value in mireds.

Reply:
I assume you know what you are doing ;)